src/components/domain/auth/SignInWithPhone.tsx

```diff
@@ -35,7 +39,37 @@ export const SignInWithPhone = () => {
 
   const handleSubmitCode = async () => {
     const confirmationResult = window.confirmationResult;
-    await confirmationResult.confirm(code);
+    await confirmationResult.confirm(code).then(result => {
+      // IMPORTANT: Since Firebase's cloud functions doesn't have authentication triggers when signing in with credentials;
+      // the backend can't generate the initial data, as a workaround, the front can take the time diff between the creation
+      // time and the current time. If it's less than 30s, it's probably a new user, so I judge that there's no problem in
+      // allow a data creation on every login in the first 30s.
+      if (
+        new Date().getTime() -
+          new Date(result.user.metadata.creationTime!).getTime() <
+        30000
+      ) {
+        const firestore = useFirestore();
+        // Configuring docs for the new user
+        try {
+          setDoc(doc(firestore, 'users', result.user.uid), {
+            email: result.user.email,
+            uid: result.user.uid,
+          });
+
+          setDoc(doc(firestore, 'investments', result.user.uid), {
+            investedAmount: 0,
+            stocks: [],
```

Comment:
Create a function to generate these deps and reuse in the one tap login callback as well

Reply:
Done!